diagnostics/diagnostics.go

```diff
@@ -27,28 +27,30 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pilosa/pilosa"
 	"github.com/shirou/gopsutil/host"
 	"github.com/shirou/gopsutil/mem"
 	"github.com/sony/gobreaker"
 )
 
+// Ensure ActiveDiagnostics implements interface.
+var _ pilosa.Diagnostics = &ActiveDiagnostics{}
+
 // TODO: unique Cluster ID
 
 // Default version check URL.
 const (
-	DefaultVersionCheckURL = "https://diagnostics.pilosa.com/v0/version"
+	defaultVersionCheckURL = "https://diagnostics.pilosa.com/v0/version"
 )
 
 type versionResponse struct {
 	Version string `json:"version"`
 	Message string `json:"message"`
 }
 
-// Diagnostics represents a client to the Pilosa cluster.
-type Diagnostics struct {
+// ActiveDiagnostics represents a client to the Pilosa cluster.
+type ActiveDiagnostics struct {
```

Comment:
this naming breaks the stuttering rule (since the package is now diagnostics, when pilosa/server imports and uses it, we get `diagnostics.NewActiveDiagnostics`. The solution here is not to break up our packages by functionality, but by dependency. So the subpackage `diagnostics` should actually be named `gopsutil`, and contain a struct called `Diagnostics` and a `NewDiagnostics` constructor (if necessary). This, of course, is not my original idea, but @benbjohnson's (or at least that's where I learned it), but it seems to work quite well in practice, and I've had some success with it in the PDK. Then if we wanted another implementation of diagnostics, it would live in another package named by dependency, or if it had no external dependencies (i.e. if it was a stub or mock implementation, it could live in `pilosa/diagnostics.go` alongside the interface definition.